lib/logstash/outputs/email.rb

```diff
@@ -137,18 +143,31 @@ def receive(event)
       end
       mail.cc = event.sprintf(@cc)
       mail.subject = formatedSubject
-      if @htmlbody.empty?
+
+      if @htmlbody.empty? and @template_file.empty?
         formattedBody.gsub!(/\\n/, "\n") # Take new line in the email
         mail.body = formattedBody
       else
+        # This handles multipart emails
+        # cf: https://github.com/mikel/mail/#writing-and-sending-a-multipartalternative-html-and-text-email
         mail.text_part = Mail::Part.new do
           content_type "text/plain; charset=UTF-8"
           formattedBody.gsub!(/\\n/, "\n") # Take new line in the email
           body formattedBody
         end
-        mail.html_part = Mail::Part.new do
-          content_type "text/html; charset=UTF-8"
-          body formattedHtmlBody
+        
+        if @template_file.empty?
+          mail.html_part = Mail::Part.new do
+            content_type "text/html; charset=UTF-8"
+            body formattedHtmlBody
+          end
+        else
+          htmlTemplate = File.open(@template_file, "r").read
```

Comment:
this should be moved to the `register` method like:

```ruby
  @htmlTemplate = File.open(@template_file, "r").read unless @template_file.empty?
```

Otherwise we'll be opening the template file for each event.

Reply:
Good catch!